pandas/_libs/lib.pyx

```diff
@@ -752,8 +752,11 @@ cpdef ndarray[object] ensure_string_array(
             out = arr.astype(str).astype(object)
             out[arr.isna()] = na_value
             return out
-
-        arr = arr.to_numpy()
+        if hasattr(arr, "type"):
+            # pyarrow array
+            arr = np.array(arr)
```

Comment:
1) can we handle this at a higher level?  2) why doesnt the pyarrow object's to_numpy work here?

Reply:
It sets zero_copy_only to True which causes raising here.

we have to do a check like this somewhere as long as arrow is not a hard dependency, can do it at a higher level as well

Reply:
moved it to a higher level, not sure if it is better though